xarray/core/datatree.py

```diff
@@ -158,6 +158,18 @@ def _check_alignment(
             _check_alignment(child_path, child_ds, base_ds, child.children)
 
 
+def _check_for_slashes_in_names(variables: Iterable[Hashable]) -> None:
+    offending_variable_names = [
+        name for name in variables if isinstance(name, str) and "/" in name
+    ]
+    if len(offending_variable_names) > 0:
+        raise KeyError(
+            f"Given Dataset contains path-like variable names: {offending_variable_names}. "
+            "A Dataset represents a group, and a single group "
+            "cannot have path-like variable names. "
```

Comment:
Nice error message, but it should probably also mention that it is the presence of the `/` character specifically that is the issue.

Reply:
Indeed, over-abstraction in error messages is often painful for the users reading them, me included. I updated the error message to explicitly mention `/`.